staging/src/k8s.io/api/resource/v1beta1/types.go

```diff
@@ -1387,6 +1387,12 @@ type DeviceClassSpec struct {
 	// it got removed. May be reused once decoding v1alpha3 is no longer
 	// supported.
 	// SuitableNodes *v1.NodeSelector `json:"suitableNodes,omitempty" protobuf:"bytes,3,opt,name=suitableNodes"`
+
+	// ExtendedResourceName is the extended resource name for the devices of this class.
+	// This is an alpha field.
+	// +optional
+	// +featureGate=DRAExtendedResource
```

Comment:
Feature gated fields must be dropped if the feature gate is disabled and the field is not set already.

See https://github.com/kubernetes/kubernetes/blob/9822e51403e46cc0aeae49cf64eae653eff29f9e/pkg/registry/resource/resourceclaim/strategy.go#L212-L345 for how that is done for ResourceClaim. You need to add similar code to https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/resource/deviceclass/strategy.go and to https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/strategy.go for `ExtendedResourceClaimStatus` - currently both is missing.

Reply:
added strategy